core/src/main/scala/kafka/server/AbstractFetcherThread.scala

```diff
@@ -17,19 +17,20 @@
 
 package kafka.server
 
+import java.util.concurrent.locks.ReentrantLock
+
 import kafka.cluster.BrokerEndPoint
-import kafka.utils.{Pool, ShutdownableThread}
-import kafka.consumer.{PartitionTopicInfo, SimpleConsumer}
-import kafka.api.{FetchRequest, FetchResponse, FetchResponsePartitionData, FetchRequestBuilder}
-import kafka.common.{KafkaException, ClientIdAndBroker, TopicAndPartition, ErrorMapping}
-import kafka.utils.DelayedItem
-import kafka.utils.CoreUtils.inLock
-import kafka.message.{InvalidMessageException, ByteBufferMessageSet, MessageAndOffset}
+import kafka.consumer.PartitionTopicInfo
+import kafka.message.{InvalidMessageException, MessageAndOffset, ByteBufferMessageSet}
+import kafka.utils.{Pool, ShutdownableThread, DelayedItem}
+import kafka.common.{KafkaException, ClientIdAndBroker, TopicAndPartition}
 import kafka.metrics.KafkaMetricsGroup
-
-import scala.collection.{mutable, Set, Map}
+import kafka.utils.CoreUtils.inLock
+import org.apache.kafka.common.protocol.Errors
+import AbstractFetcherThread._
+import scala.collection.{JavaConverters, mutable, Set, Map}
+import JavaConverters._
```

Comment:
I found that you put forward the bug kafka-1711, where should i put the file? what's the direction. I hava download the file ConsoleProducer.scala,and also change the file accordint to the order, 2 line deleted and 4 line added.